### PR TITLE
fix: move duplication to backend

### PR DESCRIPTION
**Pull request by rkorzhoff**
- Task duplication moved to backend
- (feat) Agent selector added to task
- Add textarea to logs

## Summary by CodeRabbit

- **New Features**
	- Introduced an `AgentSelector` component for selecting agents from a dropdown.
	- Added task duplication functionality through a new `duplicateTask` API.
	- New UI enhancements including a `TaskStatusBadge` component and improved chat input features.
	- Implemented agent selection in task forms and full task views.
- **Enhancements**
	- Updated font styling in `TaskInput` for better readability.
	- Improved task status display to use sentence case.
	- Enhanced task grouping logic to exclude empty groups.
	- Modified task navigation to handle creation flags more flexibly.
- **Bug Fixes**
	- Adjusted `TaskControls` to directly await task execution results.
	- Fixed the scrolling behavior in `TaskLogs` for better message viewing.
- **Style**
	- Updated icon sizes in `TaskStatusInProgress` for better visibility.
	- Enhanced chat and task UI components with new styling and layout adjustments.
- **Documentation**
	- No visible changes to end-users.
- **Refactor**
	- Streamlined imports and component usage across various modules for efficiency.
- **Tests**
	- No visible changes to end-users.
- **Chores**
	- Reordered component exports for consistency.
- **Revert**
	- No visible changes to end-users.

### CODE REVIEW

File: features/task/lib/groupTasks.ts [hunk 1]
```diff
@@ -33,6 +33,7 @@ export const groupTasks = (tasks: Task[]): GroupedTasks => {
     const statusGroup = getStatusGroup(task.status)
     groupedTasks[statusGroup].push(task)
   })
+  // remove empty groups
```
[the remark this answers is not in the document]
Comment (rkorzhoff) [reply features/task/lib/groupTasks.ts]: its unnecessary, I've removed that comment already

File: src-tauri/src/repo/tasks.rs [hunk 2]
```diff
@@ -385,7 +386,8 @@ pub async fn update<'a, E: Executor<'a, Database = Sqlite>>(
         SET
             title = COALESCE($1, title),
             summary = COALESCE($2, summary),
-            updated_at = $3
+            updated_at = $3,
+            agent_id = $5
         WHERE id = $4
```
Comment (ri-nat) [inline src-tauri/src/repo/tasks.rs:391]: It would be better to have them properly ordered. Just easier to maintain in the future.

**Review by ri-nat — CHANGES_REQUESTED** [review]
Everything else looks great 👍 May be considered approved after the fix.

File: src-tauri/src/repo/tasks.rs (hunk 2 above)
Comment (rkorzhoff) [reply src-tauri/src/repo/tasks.rs]: Done

**Review by botjoker — DISMISSED** [review]

**Review by ri-nat — APPROVED** [review]

**Pull request merged**